### ISSUE TITLE
instrumentationKey set later

### ISSUE THREAD

**Issue by jycouet** [MarkPieszak/angular-application-insights#17]
Hi, 

I would like to know if it's possible to set the instrumentationKey later and not in @NgModule?
```
ApplicationInsightsModule.forRoot({
    instrumentationKey: 'INSERT_instrumentationKey_HERE'
})
```
(and not having the log: "An instrumentationKey value is required to initialize AppInsightsService")

Because depending on my environement I would like to set different "instrumentationKey".

Thank you

**Comment by MarkPieszak** [comment]
If possible you could either pluck it from an object on window, or from a static property of some Class. If you don't get it until some http call finishes you could use APP_INITIALIZER as well. How are you setting this property, is just another .ts file?

**Comment by jycouet** [comment]
Thank you for your answer.
Actually, I put now the key in **environment.prod.ts**. Like this I can manage different environments.

Now the only stuff I don't know is: Should I commit these environment files in git? Since they contains "keys"?

**Comment by MarkPieszak** [comment]
I don't think it'd be any harm to have them in there, you'd need to have the keys visible/public either way (whether you typed them in there or not). You should be fine committing them I'd imagine!

**Comment by saskodh** [comment]
@MarkPieszak, can you show an example how you can configure it with an APP_INITIALIZER which retrieves the key from the backend? I think in order to support that the instrumentationKey config should also take a function which will be called to retrieve the key,

**Comment by MarkPieszak** [comment]
@saskodh fixed via  2c6c713 thanks to @jycouet
